lib/pbench/server/filetree.py

```diff
@@ -299,18 +299,22 @@ def unpack(self, incoming: Path, results: Path):
         ensured by calling this indirectly through the Controller class unpack
         method.
 
-        TODO: This is a prototype for testing, and doesn't actually unpack
-        the tarball as we're going to be relying on the current unpack pipeline
-        command for some time.
-
         Args:
             incoming: Controller's directory in the INCOMING tree
             results: Controller's directory in the RESULTS tree
         """
-        self.logger.warning("Tarball unpack is not yet implemented")
         unpacked = incoming / self.name
-        unpacked.mkdir(parents=True)  # Just create an empty directory for now
+        unpacked.mkdir(parents=True)
         self.unpacked = unpacked
+
+        # extracting Tarball in unpacked directory
+        try:
+            tarball = tarfile.open(self.tarball_path)
+            tarball.extractall(self.unpacked)
```

Comment:
Pete has expressed concerns in the past about the performance of the Python `tarfile` package vs spawning the `tar` command as a child process.

Logically I don't necessarily object to this, and changing back to `tar` would be completely encapsulated under this method, but it's something to think about.

Reply:
I do recall playing around with the python `tarfile` package based on Peter's comment about this in the past when I was working on tar command re-try in the TM. And it does seem to have performance lag very slightly compared to just running a `tar` command as a child process but I didn't take other factors into consideration that might have caused it so I cant say for sure.

Reply:
Yes, wouldn't it be interesting to run real side-by-side performance comparisons? Perhaps someday someone will find time to do that! (We really will need to run serious performance analysis on our server; and ideally build performance regression testing into our CI pipeline...)

Reply:
It is all about size of tar ball.  Get's worse as size grows.

Reply:
After going through spawning I had some questions: do we want unpacking to be done as a daemon process or the main process should wait for it to end? Also should I perform spawning using `multiprocess.Process` or `distutils.spawnl`? `distutils.spawnl` have been used by peter in `tool_meister_start`, so, I just thought of asking.